### PR TITLE
Fix merge conflict recovery UX

### PR DESCRIPTION
**Pull request by wwwillchen**
## Summary

Make AI-assisted merge-conflict recovery a clear, resumable sync flow instead of leaving stale controls behind after the chat resolves conflicts.

- Keep users in the conflict-resolution chat while the GitHub panel remains available, and explicitly track resolving, verifying, and ready-to-sync phases in the main-owned GitHub operation state.
- Correlate the recovery state with the exact main-owned chat, so completion survives navigation or a renderer remount instead of depending on a window-local callback.
- Replace the stale AI action with **Continue to Sync** only after Git confirms the conflicts are gone; continuing resumes the interrupted push, including `rebase --continue` before the final push when needed.
- Remove the duplicate merge-conflict error banner so the dedicated recovery surface is the single source of guidance.
- Restyle the recovery surface as a compact, neutral status panel with clearer hierarchy, file rows, state-specific icons, and concise actions inspired by GitHub/Linear/Vercel patterns.
- Preserve the in-progress phase across ambient repository reconciliation, so background probes cannot incorrectly reset the workflow while the AI chat is still working.
- If repository verification fails, keep the resolved work intact and show one inline **Try again** action that retries verification without rerunning the AI chat.
- Preserve the sanitized verification failure inline, and require the explicit retry action so focus-driven repository reconciliation cannot silently clear or restart the failed phase.
- Keep cancel and branch-switch escape hatches available in stable recovery phases, while intentionally blocking them during active AI edits and Git verification to avoid racing repository mutations.
- Notify users when a foreground sync first pauses for conflicts, while avoiding a duplicate persistent error banner and keeping ambient conflict discovery quiet.
- Announce recovery-phase changes through a polite live region for screen-reader users.
- Correlate completion with the exact resolution chat and each verification attempt, so delayed events from earlier work cannot advance or fail the current recovery.
- Use the retained chat snapshot as the single completion source, avoiding duplicate completion dispatches from both the hook callback and connector effect.
- Tell the AI to edit conflicted files in place and verify that no markers remain, rather than merely describing or pasting a resolution in chat.
- Keep the scope intentionally narrow: this does not redesign the broader GitHub settings panel.

<a href="https://cubic.dev/pr/dyad-sh/dyad/pull/4202?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

---

> [!NOTE]
> **Medium Risk**
> Touches main-owned GitHub sync state machine, probe failure routing, and push/rebase continuation after conflicts—important for data integrity but heavily tested and scoped to conflict recovery.
> 
> **Overview**
> **GitHub merge/sync conflict recovery** is now a phased, main-owned workflow instead of treating a clean ambient probe as “done” or leaving a stale **Resolve with AI** control after chat finishes.
> 
> The `github_ops` **conflicted** state gains optional `resolution` phases (`resolving` → `checking` → `ready-to-sync` or `verification-failed`), chat correlation (`resolutionChatId`), and **verification attempt** tokens so stale probe/chat events cannot advance or fail the wrong recovery. New events include `CONFLICT_RESOLUTION_STARTED` / `FINISHED` (with `chatId`), `CONFLICT_VERIFICATION_FAILED`, and `RETRY_CONFLICT_VERIFICATION`; probe commands carry `verificationAttempt`. After Git confirms no markers, users get **Continue to Sync** (push or `rebase-continue` then push) instead of restarting AI. Verification errors surface inline with **Try again** (no error toast during dedicated checking). Conflict entry drops the duplicate `MERGE_CONFLICT` error banner in favor of a single info-style recovery panel.
> 
> **GitHubConnector** watches the correlated chat stream and dispatches `CONFLICT_RESOLUTION_FINISHED` when the resolution chat completes, and renders the new status panel (file list, spinners, continue/retry/cancel). **Capabilities** gate resolve/cancel/continue/retry by phase. **Main command runner** routes probe failures during `checking` to `CONFLICT_VERIFICATION_FAILED` instead of generic toasts. **AI prompt** instructs in-place file edits and marker verification. Tests and `state-machines.md` document the in-progress vs checking probe rule and finite graph exploration for retry counters.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit b36b9d76afee7374ec63af7046192d6d6ea1f1d1. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/ipc/handlers/__tests__/git_collaboration.integration.test.tsx [hunk 1]
```diff
@@ -49,6 +52,26 @@ function git(appDir: string, ...args: string[]): string {
...
+      "-c",
+      "commit.gpgsign=false",
+      ...args,
+    ],
+    {
+      cwd: appDir,
+      env: { ...process.env, GIT_TERMINAL_PROMPT: "0" },
+    },
+  );
+  return result.stdout;
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/ipc/handlers/__tests__/git_collaboration.integration.test.tsx:72]: This does not apply here: Node’s `execFile` uses UTF-8 string output by default unless an option requests Buffer output. The promisified overload therefore returns `stdout: string`, which is also enforced by the passing project type-check. No conversion is needed.

**Comment by wwwillchen** [comment]
## 🤖 Claude Code Review Summary

### PR Confidence: 3/5
All trusted review threads are resolved and focused tests pass, but the independent deep review still found one non-threaded recovery edge case when the post-resolution conflict probe itself fails.

### Unresolved Threads

_No unresolved threads_

### Resolved Threads

| Issue | Rationale | Link |
|-------|-----------|------|
| Ambient probes could finish AI resolution early | The machine now holds `resolving` until the explicit chat completion event, then lets only the `checking` probe advance recovery. | [Codex](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724071425), [Copilot](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724118702), [Cubic](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724160117) |
| Rebase conflicts resumed with a push | Rebase-origin recovery now runs `rebase-continue` and chains the pending normal push, with actor and transition coverage. | [Codex](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724071435), [Cubic](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724160087) |
| Continuation policy was duplicated | Projection and transitions now share the same continuation selector. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724160091) |
| Verification phase lacked explicit UX and selector coverage | The panel now says “Verifying resolution…” and the capability matrix covers `checking` with recovery actions disabled. | [UX](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724160104), [Capabilities](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724160097) |
| Merge-conflict fallback error could be hidden | The duplicate banner stays suppressed when the dedicated panel exists, but remains visible if recovery state was not established. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724160111) |
| Recovery graph omitted completion | Exhaustive exploration now includes the completion event and covers `checking`/`ready-to-sync`. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724160126) |
| Integration test stopped at push receipt | It now settles in-flight work, fetches the remote branch, verifies local and remote commits match, and checks the remote file has no conflict markers. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724160133) |
| `execFile` output was reported as a Buffer | Resolved with an explanation: the default encoding produces a string, as confirmed by the project type-check. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724118668) |

---
🤖 Generated by Claude Code

File: src/github_ops/transition.ts [hunk 2]
```diff
@@ -55,6 +61,19 @@ function requestOperation(
     case "running":
       return ignore(state, "op-in-flight");
     case "conflicted":
+      if (state.resolution === "resolving" || state.resolution === "checking") {
```
[the remark this answers is not in the document]
Comment (wwwillchen) [reply src/github_ops/transition.ts:70]: Addressed in b36b9d76a: cancel and branch-switch escape hatches now remain available during the stable verification-failed and ready-to-sync phases. The active resolving/checking phases intentionally remain locked because aborting or switching while AI edits or Git probes are running would race repository mutations. Per **Principle #4: Transparent Over Magical**, the UI exposes the safe recovery choices without pretending an unsafe concurrent cancellation is reliable.

**Comment by wwwillchen** [comment]
## 🤖 Claude Code Review Summary

### PR Confidence: 5/5
All trusted review threads are resolved, the recovery behavior is covered by focused state-machine and actor tests, and formatting, linting, and type-checking pass.

### Unresolved Threads

_No unresolved threads_

### Resolved Threads

| Issue | Rationale | Link |
|-------|-----------|------|
| Chat completion identity and duplicate settlement | Correlated completion with the exact resolution chat and made the retained chat snapshot the single completion source. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724304137) |
| Verification attempt identity, errors, and explicit retry | Correlated probe events with attempts, ignored stale results, preserved sanitized failure reasons, separated explicit retry from ambient reconciliation, and completed the retry coverage. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724366657) |
| Safe recovery escape hatches | Enabled cancel and branch switching in stable recovery phases. Per **Principle #4: Transparent Over Magical**, active edit/probe phases remain locked to avoid presenting an unsafe concurrent cancellation as reliable. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724383688) |
| Foreground sync conflict notice | Restored one informational notification for user-initiated sync conflicts while keeping the duplicate persistent error banner removed and ambient discovery quiet. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724383699) |
| Accessible phase changes | Added a polite, atomic status live region around recovery state and detail copy. | [View](https://github.com/dyad-sh/dyad/pull/4202#discussion_r3724383707) |

<details>
<summary>Product Principle Suggestions</summary>

_No suggestions_

</details>

---
🤖 Generated by Claude Code
